### ISSUE TITLE
A11y: Remove H1 option from ProductTitle block in the context of a grid.

### ISSUE THREAD

**Issue by nerrad** [woocommerce/woocommerce-gutenberg-products-block#1101]
For a11y reasons, I think we should consider removing H1 from the options for `ProductTitle` component in the context of the grid?  Probably should explore _what_ options should be available.  We could also look into giving an a11y notice when user's select a heading level that would break a11y guidelines in content instead?
